Clarify that the results of division operations on integer array data types resulting in floating-point outputs is implementation-defined (#362)
data-apis/array-api: spec/API_specification/array_object.md

```{note}
If both `self` and `other` have integer data types, the result of `__pow__` when `other_i` is negative (i.e., less than zero) is unspecified and thus implementation-dependent.

> If `self` has an integer data type and `other` has a floating-point data type, behavior is implementation-dependent. This is because type promotion between data type “kinds” (i.e. integer versus floating-point) is unspecified.

Reviewer: ```suggestion
If `self` has an integer data type and `other` has a floating-point data type, behavior is implementation-dependent, as type promotion between data type "kinds" (e.g., integer versus floating-point) is unspecified.
```

Evaluates `self_i / other_i` for each element of an array instance with the respective element of the array `other`.

```{note}

> If one or both of `self` and `other` have integer data types, the result is implementation-dependent. This is because type promotion between data type “kinds” (i.e. integer versus floating-point) is unspecified.

Reviewer: ```suggestion
If one or both of `self` and `other` have integer data types, the result is implementation-dependent, as type promotion between data type "kinds" (e.g., integer versus floating-point) is unspecified.
```

> Specification-compliant libraries may choose to raise an error or return an array containing the element-wise results. If an array is returned, it must have a floating-point data type.

Reviewer: Here the distinction of "result" is the actual values returned which is separate from the requirement that the output dtype must be floating-point. What this note is getting at is that, because implementations may vary in how they promote between different "kinds", what values are returned may also vary, but the output result dtype must be floating-point regardless.

> Specification-compliant libraries may choose to raise an error or return an array containing the element-wise results. If an array is returned, it must have a floating-point data type.

Reviewer: ```suggestion
Specification-compliant libraries may choose to raise an error or return an array containing the element-wise results. If an array is returned, the array must have a floating-point data type.
```

> Specification-compliant libraries may choose to raise an error or return an array containing the element-wise results. If an array is returned, it must have a floating-point data type.

Reviewer: I'll mark this conversation as resolved.
